### PR TITLE
Fix UPDATE_NEEDED is sent if user requested PTU from HMI

### PR DESCRIPTION
**Pull request by LitvinenkoIra**
Add is_notification_needed flag as a parameter to ForcePTExchange function, 
because according to EXTERNAL_PROPRIETARY exception: No notification should be sent on user requested PTU from HMI (via SDL.UpdateSDL request).

### CODE REVIEW

**Comment by LitvinenkoIra** [comment]
@LevchenkoS @okozlovlux please review

File: src/components/application_manager/test/policy_handler_test.cc [hunk 1]
```diff
@@ -85,6 +85,10 @@ using ::testing::DoAll;
 using ::testing::SetArgReferee;
 using ::testing::Mock;
 
+namespace {
+bool is_notification_needed = true;
```
Comment (okozlovlux) [inline src/components/application_manager/test/policy_handler_test.cc]: @LitvinenkoIra , kIsNotificationNeeded

File: src/components/application_manager/test/policy_handler_test.cc [hunk 2]
```diff
@@ -865,7 +868,10 @@ TEST_F(PolicyHandlerTest, PTExchangeAtUserRequest) {
   // Arrange
   ChangePolicyManagerToMock();
   // Check expectations
-  EXPECT_CALL(*mock_policy_manager_, ForcePTExchange());
+#ifdef EXTERNAL_PROPRIETARY_MODE
+  bool is_notification_needed = false;
+#endif
+  EXPECT_CALL(*mock_policy_manager_, ForcePTExchange(is_notification_needed));
```
Comment (okozlovlux) [inline src/components/application_manager/test/policy_handler_test.cc]: @LitvinenkoIra , can this expect_call be in `#ifdef EXTERNAL_PROPRIETARY_MODE`

File: src/components/policy/policy_external/test/policy_manager_impl_ptu_test.cc [hunk 3]
```diff
@@ -51,6 +51,10 @@ using ::testing::_;
 using ::testing::Return;
 using ::testing::ReturnRef;
 
+namespace {
+bool is_notification_needed = true;
```
Comment (okozlovlux) [inline src/components/policy/policy_external/test/policy_manager_impl_ptu_test.cc]: @LitvinenkoIra , please name it like global cons. see comment above.
Check other tests

File: src/components/application_manager/test/policy_handler_test.cc (hunk 1 above)
Comment (LitvinenkoIra) [inline src/components/application_manager/test/policy_handler_test.cc]: @okozlovlux Done 9677037805844ec491bcbe1b54ae7d7872113825

File: src/components/application_manager/test/policy_handler_test.cc (hunk 2 above)
Comment (LitvinenkoIra) [inline src/components/application_manager/test/policy_handler_test.cc]: @okozlovlux Done 9677037805844ec491bcbe1b54ae7d7872113825

File: src/components/policy/policy_external/test/policy_manager_impl_ptu_test.cc (hunk 3 above)
Comment (LitvinenkoIra) [inline src/components/policy/policy_external/test/policy_manager_impl_ptu_test.cc]: @okozlovlux Done 9677037805844ec491bcbe1b54ae7d7872113825

**Comment by LitvinenkoIra** [comment]
Rebuild required

**Comment by LuxoftAKutsan** [comment]
@LitvinenkoIra @okozlovlux @LevchenkoS  please review. Implementation changed

**Comment by LuxoftAKutsan** [comment]
@jacobkeeler style fixed with 6c6e347 ( will be rebased after review) 
Solution is rely hacked-up. But I still can't find better one.
Main problem that it is the only one case when we should not send notification about update needed on HMI. 
This notification is sends in callback of starting PT exchange. And to avoid sending this notification we need have information about trigger of UPDATE_NEEDED state in callback. It can be done with member flag or passing flag throw functions parameter (need to be changed about 7 interfaces)

I will communicate with colleagues for searching better one solution.

**Comment by LuxoftAKutsan** [comment]
@AGaliuzov @dev-gh  Do you have any ideas according this fix?

**Comment by LuxoftAKutsan** [comment]
@LitvinenkoIra @okozlovlux @LevchenkoS please review. Implementation changed
@jacobkeeler  please also take a look on new solution.

File: src/components/policy/policy_external/include/policy/update_status_manager.h [hunk 4]
```diff
@@ -151,6 +151,13 @@ class UpdateStatusManager {
   void ScheduleUpdate();
 
   /**
+   * @brief ScheduleUpdate allows to schedule next update.
+   * It will change state to Update_Needed, that's is
+   * and will not send and notifications about updating to HMI
```
Comment (okozlovlux) [inline src/components/policy/policy_external/include/policy/update_status_manager.h]: @LuxoftAKutsan , fix error in this comment

File: src/components/utils/test/test_generator/generated_msg_version_test.cc [hunk 5]
```diff
@@ -56,7 +58,7 @@ void get_version_from_file(int& first_version, int& second_version, int& third_v
     std::size_t isfound = str.find("interface name");
     if (isfound != std::string::npos) {
       std::size_t versionStart = str.find("\"", str.find("version="));
-      ASSERT_TRUE(versionStart != std::string::npos);  
+      ASSERT_TRUE(versionStart != std::string::npos);
```
Comment (jacobkeeler) [inline src/components/utils/test/test_generator/generated_msg_version_test.cc]: Sorry, these style issues were introduced in a previous PR, they are now fixed by #1534. They should be removed from this PR before merge

File: src/components/policy/policy_external/src/update_status_manager.cc [hunk 6]
```diff
@@ -62,6 +62,8 @@ UpdateStatusManager::~UpdateStatusManager() {
 void UpdateStatusManager::ProcessEvent(UpdateEvent event) {
   sync_primitives::AutoLock lock(status_lock_);
   current_status_->ProcessEvent(this, event);
+  last_processed_event_ = event;
+  LOG4CXX_DEBUG(logger_, "last_processed_event_ = " << last_processed_event_);
```
Comment (jacobkeeler) [inline src/components/policy/policy_external/src/update_status_manager.cc]: This duplicate log (appears in `DoTransition()` as well) might cause problems with debugging, should probably be removed

File: src/components/policy/policy_external/src/update_status_manager.cc [hunk 7]
```diff
@@ -184,14 +190,20 @@ void UpdateStatusManager::DoTransition() {
 
   current_status_ = next_status_;
   next_status_.reset();
-  listener_->OnUpdateStatusChanged(current_status_->get_status_string());
+
+  LOG4CXX_DEBUG(logger_, "last_processed_event_ = " << last_processed_event_);
+  if (last_processed_event_ != kScheduleManualUpdate) {
+    listener_->OnUpdateStatusChanged(current_status_->get_status_string());
+  }
```
Comment (jacobkeeler) [inline src/components/policy/policy_external/src/update_status_manager.cc]: The one difference I see with this solution is that this line will no longer be called:
```
policy_manager_->SaveUpdateStatusRequired(policy::kUpToDate != status);
```
Would it cause any problems not updating the policy table's `update_needed` status?

Comment (LuxoftAKutsan) [inline src/components/policy/policy_external/src/update_status_manager.cc]: @jacobkeeler this method is redundant. Policy table should be saved to file system on shutdown any way. 
And during ignition cycle SDL use update_status manager for checking current state.

File: src/components/policy/policy_external/src/update_status_manager.cc (hunk 6 above)
Comment (LuxoftAKutsan) [inline src/components/policy/policy_external/src/update_status_manager.cc]: @jacobkeeler removed this debug logs.

File: src/components/utils/test/test_generator/generated_msg_version_test.cc (hunk 5 above)
Comment (LuxoftAKutsan) [inline src/components/utils/test/test_generator/generated_msg_version_test.cc]: @jacobkeeler removed style changes from  src/components/utils/test/test_generator/generated_msg_version_test.cc

File: src/components/application_manager/include/application_manager/policies/policy_handler.h [hunk 8]
```diff
@@ -121,7 +121,6 @@ class PolicyHandler : public PolicyHandlerInterface,
   void GetUpdateUrls(const uint32_t service_type,
                      EndpointUrls& out_end_points) OVERRIDE;
   virtual std::string GetLockScreenIconUrl() const OVERRIDE;
-  void ResetRetrySequence() OVERRIDE;
```
Comment (jacobkeeler) [inline src/components/application_manager/include/application_manager/policies/policy_handler.h]: Why the removal of this function?

File: src/components/policy/policy_regular/include/policy/policy_manager.h [hunk 9]
```diff
@@ -1,486 +0,0 @@
...
-   */
-  virtual void StartPTExchange() = 0;
-};
-
-}  // namespace policy
-
-extern "C" policy::PolicyManager* CreateManager();
-extern "C" void DeleteManager(policy::PolicyManager*);
-
-#endif  // SRC_COMPONENTS_POLICY_POLICY_REGULAR_INCLUDE_POLICY_POLICY_MANAGER_H_
```
Comment (jacobkeeler) [inline src/components/policy/policy_regular/include/policy/policy_manager.h]: Why was this file removed?

Comment (jacobkeeler) [inline src/components/policy/policy_regular/include/policy/policy_manager.h]: This was also removed in #1478, this PR could be rebased once that PR is merged which would resolve the issue.

File: src/components/application_manager/include/application_manager/policies/policy_handler.h (hunk 8 above)
Comment (LuxoftAKutsan) [inline src/components/application_manager/include/application_manager/policies/policy_handler.h]: @jacobkeeler changes removed

Comment (LuxoftAKutsan) [inline src/components/application_manager/include/application_manager/policies/policy_handler.h]: @jacobkeeler this method is unused. Remiving this method extracted to separate commit 0682954

File: src/components/policy/policy_regular/include/policy/policy_manager.h (hunk 9 above)
Comment (LuxoftAKutsan) [inline src/components/policy/policy_regular/include/policy/policy_manager.h]: @jacobkeeler rebased on latest develop
